providers/dns/nearlyfreespeech/internal/client.go

```diff
@@ -32,8 +32,21 @@ type Client struct {
 	HTTPClient *http.Client
 }
 
+// Exposed separately so that tests can share the workaround below.
+func parseURL(uri string) *url.URL {
+	baseURL, err := url.Parse(uri)
+	if err != nil {
+		panic(err)
+	}
+	// Workaround for https://golang.org/issue/58605
+	if baseURL.Path == "" {
+		baseURL.Path = "/"
+	}
+	return baseURL
+}
```

Comment:
this function is useless because the endpoints are always called with a path after the baseURL.

Reply:
Except in tests.

Reply:
It's useless outside the tests, so this must be moved inside the tests.

Reply:
I need to do the same thing (fix up `Path`):
 - in test,
 - in NewClient.

Where would you propose I put this function?

Reply:
I propose to remove this function, add alter the path inside the set up of the client.

Reply:
I need to read more, I feel like it's also useless inside the tests.

Reply:
Doing the `Path`-mangling *in NewClient* is what fixes the bug.